### PR TITLE
fix(tables): compare order_key bytewise (COLLATE "C") to stop insert collation errors

### PR DESCRIPTION
**Pull request by TheodoreSpeaks**
## Summary
- Pin `user_table_rows.order_key` to `COLLATE "C"` (migration 0228) so Postgres compares fractional-index keys bytewise, matching the in-house fractional-indexing library. The prod DB default `en_US.UTF-8` sorts lowercase before uppercase (the opposite of the `Z < a` ASCII boundary the keys rely on), so `max()`/neighbor lookups returned out-of-order bounds and `generateKeyBetween` threw `a >= b` on insert (and rows displayed out of order). Local DBs default to `C.UTF-8`, so it never reproduced locally.
- Add a one-off repair script `repair-table-order-key-collation.ts` that re-keys any table with bytewise inversions/duplicates from `(position, id)` order, under the per-table advisory lock (idempotent, `--dry-run`). Separate from the NULL backfill, which already ran.
- Harden `resolveInsertByNeighbor` to anchor on the next/prev *distinct* key so a stray duplicate can never throw.
- Add fractional-indexing tests covering the `Z`/`a` boundary that en_US inverts.

## Type of Change
- [x] Bug fix

## Testing
Tested manually. `bun run lint` clean, `bun run check:api-validation:strict` passed, `tsc --noEmit` clean, fractional-indexing tests (5) + table suite (196) pass, `drizzle-kit check` clean.

## Operational
Run after deploy: apply migration 0228, then `repair-table-order-key-collation.ts --dry-run` to size, then for real. Migration + neighbor fix stop the error independent of the `TABLES_FRACTIONAL_ORDERING` flag; the repair is pre-cleaning needed before the flag is enabled.

## Checklist
- [x] Code follows project style guidelines
- [x] Self-reviewed my changes
- [x] Tests added/updated and passing
- [x] No new warnings introduced
- [x] I confirm that I have read and agree to the terms outlined in the [Contributor License Agreement (CLA)](./CONTRIBUTING.md#contributor-license-agreement-cla)

### CODE REVIEW

**Comment by TheodoreSpeaks** [comment]
@greptile review

**Comment by TheodoreSpeaks** [comment]
Good catch — fixed in 075f3fec6. The detection query was ordering the window by `order_key COLLATE "C"` and then comparing consecutive keys, which is monotonic by construction, so it only ever caught duplicates. It now walks rows in authoritative `position, id` order and flags any table where `order_key` isn't strictly increasing along that walk — so swapped keys (e.g. pos 0 = `a0`, pos 1 = `Zz`) are detected, and it's still idempotent after a re-key.

File: apps/sim/scripts/repair-table-order-key-collation.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+          )
+          const rows = await trx
+            .select({ id: userTableRows.id })
+            .from(userTableRows)
+            .where(eq(userTableRows.tableId, tableId))
+            .orderBy(asc(userTableRows.position), asc(userTableRows.id))
+
+          if (rows.length === 0) return 0
+          const keys = nKeysBetween(null, null, rows.length)
+          if (dryRun) return rows.length
```
[the remark this answers is not in the document]
Comment (TheodoreSpeaks) [reply apps/sim/scripts/repair-table-order-key-collation.ts]: Fixed in e537aa593 — dry-run now skips the transaction and advisory lock entirely and does a plain `count(*)` per table outside any serializing context, so a sizing pass never queues behind live inserts.

File: apps/sim/scripts/repair-table-order-key-collation.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+              SET order_key = v.order_key
+              FROM (VALUES ${values}) AS v(id, order_key)
+              WHERE t.id = v.id AND t.table_id = ${tableId}
+            `)
+          }
+          return rows.length
+        })
+        stats.tablesKeyed += 1
+        stats.rowsKeyed += keyed
+        console.log(`  ${tableId}: re-keyed ${keyed} rows`)
```
[the remark this answers is not in the document]
Comment (TheodoreSpeaks) [reply apps/sim/scripts/repair-table-order-key-collation.ts:142]: Fixed in e537aa593. I kept the counter incrementing so the dry run still reports how many tables/rows *would* be touched (the sizing info), but relabeled everything: the per-table line now prints `would re-key`, and the summary uses `tables to re-key` / `rows to re-key` plus a `[DRY RUN — no rows written]` marker. Zeroing `tablesKeyed` would have dropped the sizing count, so I went with clear labels instead.

File: apps/sim/scripts/repair-table-order-key-collation.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            const chunk = rows.slice(start, start + WRITE_CHUNK_SIZE)
+            const values = sql.join(
+              chunk.map((r, i) => sql`(${r.id}, ${keys[start + i]})`),
+              sql`, `
+            )
+            await trx.execute(sql`
+              UPDATE user_table_rows AS t
+              SET order_key = v.order_key
+              FROM (VALUES ${values}) AS v(id, order_key)
+              WHERE t.id = v.id AND t.table_id = ${tableId}
```
[the remark this answers is not in the document]
Comment (TheodoreSpeaks) [reply apps/sim/scripts/repair-table-order-key-collation.ts:135]: Fixed in e537aa593 — the per-table log now prints `would re-key N rows` in dry-run vs `re-keyed N rows` on a live run.

File: apps/sim/scripts/repair-table-order-key-collation.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+          }
+          return rows.length
+        })
+        stats.tablesKeyed += 1
+        stats.rowsKeyed += keyed
+        console.log(`  ${tableId}: re-keyed ${keyed} rows`)
+      } catch (error) {
+        stats.failed += 1
+        console.error(`  ${tableId}: FAILED — ${getErrorMessage(error)}`)
+      }
```
[the remark this answers is not in the document]
Comment (TheodoreSpeaks) [reply apps/sim/scripts/repair-table-order-key-collation.ts:146]: Fixed in e537aa593 — the completion line appends `[DRY RUN — no rows written]` and the stat labels switch to `tables to re-key` / `rows to re-key` in dry-run.

**Comment by TheodoreSpeaks** [comment]
@greptile review
